API/Startup.cs

```diff
@@ -351,6 +366,26 @@ public void ConfigureServices(IServiceCollection services)
         });
     }
 
+    private static void UpdateBaseUrlInIndex(string baseUrl)
+    {
+        var htmlDoc = new HtmlDocument();
```

Comment:
Haha I was thinking of doing this when I was stuck as well, but seemed to be a bad idea.

Reply:
Hahaha my first thought was to edit the html response in transit every time it was requested, and then I took a step back and realized that it only needed to be changed at startup.

This was the easiest way I could find for everything to work seamlessly. Angular needed to know the `baseUrl` from the start, and there's no simple way to fetch it from the API.